sunpy/image/slit.py

```diff
@@ -0,0 +1,115 @@
+    along x.
+    
+    Parameters    
+    ----------
+    mcube_in : `sunpy.map.MapCube`
+        A mapcube of the images you want to perform the slit analysis on. 
+        Usually with x and y as space, and z as time.
+    x1, y1 : `astropy.units.Quantity` 
+        The x and y coordinates of the beginning of the slit. May be either 
+        pixel coordinates or astropy units.
+    x2, y2 : `astropy.units.Quantity`
+        The x and y coordinates of the end of the slit. May be either 
+        pixel coordinates or astropy units.
+
+        
+    Returns
+    -------
+    out : `numpy.ndarray`
+        A numpy array with intenisty in y and time in x.
+
+    """
+    
+    # check the attributes of the coordinates
+    if ((isinstance(x1 and y1, u.Quantity) and isinstance(x2 and y2, u.Quantity)) or
+        (hasattr(x1 and y1, 'unit') and hasattr(x2 and y2, 'unit'))):
+    
+        if (x1.unit.is_equivalent(mcube_in[0].units.x) and
+            y1.unit.is_equivalent(mcube_in[0].units.y)):
+            
+            # convert the world to pixel
+            init_map = sunpy.map.Map(mcube_in[0])
+            c_x1, c_y1 = init_map.data_to_pixel(x1, y1)
+            c_x2, c_y2 = init_map.data_to_pixel(x2, y2)            
+                        
+        elif x1.unit.is_equivalent(u.pixel) and y1.unit.is_equivalent(u.pixel):
+            c_x1, c_y2 = x1, y1
+            c_x2, c_y2 = x2, y2
+        
+        else:
+            raise u.UnitsError("xy1 and xy2 must be "
+                               "in units convertable to {} or {}".format(mcube_in[0].units['x'],
+                                                                         u.pixel))
+    else:
+        raise TypeError("Arguments range_a and range_b to function submap "
+                        "have an invalid unit attribute "
+                        "You may want to pass in an astropy Quantity instead.")
+        
+    
+    
+    # call to the get pixel numbers routine
+    slit = get_pixels_on_line(int(c_x1.value), int(c_y1.value), 
+                              int(c_x2.value), int(c_y2.value))    
+
+    for d_arr in mcube_in: 
+        data = d_arr.data
+        # get the initial slit pixels
+        s_values = data[slit.T[0], slit.T[1]]
+        # plus one
+        s_p1_values = data[slit.T[0], slit.T[1]]
+        # minus one
+        s_m1_values = data[slit.T[0], slit.T[1]]
+        # wap it all in one list
+        slits = [s_m1_values, s_values, s_p1_values]
+    
+    end_array = np.array(slits)
+    im_array = np.rot90(end_array)
+    return(im_array)
+
+
+
+def get_pixels_on_line(x1, y1, x2, y2, getvalues=True):
+    """
+    Uses Bresenham's line algorithm to enumerate the pixels along
+    a line.
+    (see http://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm)
+    If `getvalues`==False then it will return tuples of (x, y) coordinates
+    instead of pixel values. This was taken from the package Ginga.
+    """
```

Comment:
the docstring needs some format fixing for the url and the `getvalues==False` for nice formatting.